src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -83,15 +122,71 @@ class SetWaveClipNameCommand : public AudacityCommand
 const ComponentInterfaceSymbol SetWaveClipNameCommand::Symbol
 { XO("Set Wave Clip Name") };
 
+//Handle which is used to mouse events to TextEditHelper
+class WaveClipTitleEditHandle final : public UIHandle
+{
+    std::shared_ptr<TextEditHelper> mHelper;
+public:
+
+    WaveClipTitleEditHandle(const std::shared_ptr<TextEditHelper>& helper)
+        : mHelper(helper)
+    { }
+
+    Result Click(const TrackPanelMouseEvent& event, AudacityProject* project) override
+    {
+        if (mHelper->OnClick(event.event, project))
+            return RefreshCode::RefreshCell;
+        return RefreshCode::RefreshNone;
+    }
+
+    Result Drag(const TrackPanelMouseEvent& event, AudacityProject* project) override
+    {
+        if (mHelper->OnDrag(event.event, project))
+            return RefreshCode::RefreshCell;
+        return RefreshCode::RefreshNone;
+    }
+
+    HitTestPreview Preview(const TrackPanelMouseState& state, AudacityProject* pProject) override
+    {
+        static auto ibeamCursor =
+            ::MakeCursor(wxCURSOR_IBEAM, IBeamCursorXpm, 17, 16);
+        return {
+           XO("Click and drag to select text"),
+           ibeamCursor.get()
+        };
+    }
+
+    Result Release(const TrackPanelMouseEvent& event, AudacityProject* project, wxWindow*)
+    {
+        if (mHelper->OnRelease(event.event, project))
+            return RefreshCode::RefreshCell;
+        return RefreshCode::RefreshNone;
+    }
+
+    Result Cancel(AudacityProject* project)
+    {
+        if (mHelper)
+        {
+            mHelper->Cancel(project);
+            mHelper.reset();
+        }
+        return RefreshCode::RefreshAll;
+    }
+};
+
 WaveTrackAffordanceControls::WaveTrackAffordanceControls(const std::shared_ptr<Track>& pTrack)
     : CommonTrackCell(pTrack), mClipNameFont(wxFont(wxFontInfo()))
 {
+    if (auto trackList = pTrack->GetOwner())
+    {
+        trackList->Bind(EVT_TRACKLIST_SELECTION_CHANGE,
```

Comment:
Now that you bind to the track list, the function OnTrackChanged will take its effect whenever any track changes it selected state.  Is that the intention, or should it check only for changes of the relevant wave track?

Reply:
Yes, it's intentional, thought was not specified by design document, so I made the behaviour similar to the behaviour in the label track, whenever it was possible (that was specified)